### PR TITLE
Filter operations

### LINKED ISSUE #21: Support filter operations
We want to specify available operations as an argument to the field constructors, and have this reflected in both the spec filter utilities and the rendered spec itself.

### PR DESCRIPTION
**Pull request by ConorStevenson**
fixes #21

### CODE REVIEW

**Comment by jklepatch** [comment]
@ConorStevenson shouldnt it be the internal query that override the external one? see #f4b8eb5

**Comment by ConorStevenson** [comment]
I made a mess in git and mangled a commit with master, but it should be clearer now!

File: openapi/data/validate.py [hunk 1]
```diff
@@ -13,30 +13,39 @@ class ValidatedData:
...
-            default = field.metadata.get(DEFAULT)
-            value = data.get(field.name, default)
+            if DEFAULT in field.metadata:
+                data.setdefault(field.name, field.metadata[DEFAULT])
+
+            if field.name not in data and required and strict:
+                raise ValidationError(field.name, 'required')
+
+            ops = field.metadata.get(OPS, ())
+            for name in [field.name] + [f'{field.name}:{op}' for op in ops]:
```
Comment (lsbardel) [inline openapi/data/validate.py]: We need to use this iteration at least in two places. Lets create a function for it.
I propose something along these lines:
```python
def field_ops(field):
    yield field.name
    for op in field.metadata.get(OPS, ()):
        yield f'{field.name}:{op}'
```

File: openapi/spec/spec.py [hunk 2]
```diff
@@ -76,7 +76,9 @@ def _schema2json(self, schema):
             json_property = self._field2json(field)
             if not json_property:
                 continue
-            properties[field.name] = json_property
+            ops = field.metadata.get(OPS, ())
```
Comment (lsbardel) [inline openapi/spec/spec.py]: use `field_ops` function defined above

**Pull request merged**